open_academy/models/session.py

```diff
@@ -1,14 +1,47 @@
-from odoo import models, fields
+from datetime import date
+
+from odoo import models, fields, api, _
+
+from odoo.exceptions import ValidationError
 
 
 class Session(models.Model):
     _name = "session"
     _description = "Session"
 
     name = fields.Char(required=True)
-    start_date = fields.Date()
+    start_date = fields.Date(default=date.today())
     duration = fields.Float()
     seats = fields.Integer(string="Number of seats")
+    taken_seats = fields.Integer()
+    percent_seats = fields.Integer(string="Percentage of taken seats", compute="_compute_seats")
+    active = fields.Boolean(default=True)
     instructor_id = fields.Many2one("res.partner", domain="[('instructor', '=', True)]")
     course_id = fields.Many2one("course")
     attendee_ids = fields.Many2many("res.partner")
+
+    @api.depends("seats", "taken_seats")
+    def _compute_seats(self):
+        for record in self:
+            record.percent_seats = 0
+
+            if record.seats >= record.taken_seats and record.seats > 0:
+                record.percent_seats = round(record.taken_seats / record.seats * 100)
+
+    @api.onchange("seats", "taken_seats")
+    def _onchange_percent_seats(self):
+        if self.seats > 0:
+            self.percent_seats = round(self.taken_seats / self.seats * 100)
+
+        if self.seats > self.taken_seats or self.taken_seats < 0 or self.seats < 0:
+            return {
+                "warning": {
+                    "title": "Invalid Values",
+                    "message": "It cannot have negative number of seats nor more taken seats than total seats",
+                }
+            }
+
+    @api.constrains("instructor_id", "attendee_ids")
+    def _check_instructor_in_attendees(self):
+        if self.instructor_id not in self.attendee_ids:
+            raise ValidationError(_("Instructor not present in the attendees of his/her own session"))
```

Comment:
Do it inside a for loop, be careful when doing this kind of validations, as you will have false positives.

Reply:
Done